barretenberg/cpp/src/barretenberg/crypto/poseidon/poseidon_params.hpp

```diff
@@ -0,0 +1,508 @@
+#pragma once
+
+#include "barretenberg/ecc/curves/stark252/stark252.hpp"
+
+namespace bb::crypto {
+
+struct PoseidonStark252BaseFieldParams {
```

Comment:
It'd be good to segregate the starknet stuff in a namespace and folder - one precondition for merging this is foor it to be as clearly marked away from the audit stuff as possible

Reply:
Hey, I have made the following changes:
- Moved files under barretenberg/ext/starknet
- Moved declarations to namespace bb:starknet
Let me know if this aligns with what you had in mind